### ISSUE TITLE
Protect unpacked Flux Nodes like Inspectors are protected

### ISSUE THREAD

**Issue by SilvyPaws** [Yellow-Dog-Man/Resonite-Issues#3170]
### Is your feature request related to a problem? Please describe.

I don't think flux nodes should be grabbable and protected like inspectors are if you're not a builder. 

### Describe the solution you'd like

Protect unpacked flux nodes like inspectors are protected against non-builders.

### Describe alternatives you've considered

N/A

### Requesters

SilvyPaws

**Comment by shiftyscales** [comment]
This will just require that spawned ProtoFlux nodes are properly assigned the Developer tag like other developer interfaces, and should be a quick add. Thank you for submitting this issue. :)

**Comment by shiftyscales** [comment]
> give them a distinct "ProtoFluxNode" tag instead

That would require additional work and increase the scope of the request, @JackTheFoxOtter - as it would require retroactively applying that filter to the permission components across all existing worlds.

Overall your request sounds unrelated to this request- I'd recommend you please open a new issue if you wish for some mechanism that ProtoFlux / user-made tooling can use to identify and manipulate ProtoFlux nodes.

**Comment by JackTheFoxOtter** [comment]
It's relevant, because slots can only have one tag, and this would prevent a different tag from being used.

Actually I'm pretty sure this would break RedPrint (and potentially other node managing tools), as they explicitly are configured to NOT snap anything tagged "Developer" as to not steal inspectors. I'd make sure to test this first.

**Comment by JackTheFoxOtter** [comment]
Yes I did quickly check - if you use the "Developer" tag on ProtoFlux nodes this will break every RedPrint in the game, as the nodes won't snap to them anymore. Both the current version up to the oldest version I could find in my inventory - this would affect thousands of saved objects and worlds. So you'll kind of have to use a separate tag unless you want to risk that much content breakage.

**Comment by Stellanora64** [comment]
I also believe just changing the tag to "developer" would break @RyuviTheViali 's grid snapping protoflux tip, although they would need to confirm on that.

**Comment by JackTheFoxOtter** [comment]
I also think this would probably have to apply to existing ProtoFlux nodes as well, otherwise every "old" piece of code would still be grabbable by non-builders, while "new" code wouldn't, which would create a lot of confusing behavior.

**Comment by ProbablePrime** [comment]
Woops, Froox already had this one. ^_^

**Comment by SilvyPaws** [comment]
If adding a developer tag isn't the solution to this because it can break user content, and for some reason a Flux tag can't be added that does what Developer tag does, i'd suggestion a dynamic component that is added to unpacked flux and removed when packed that can protect unpacked Flux. It doesn't have to be retroactively added to already unpacked flux (Though if froox thinks it should, then I wont be apposed).

**Comment by ProbablePrime** [comment]
We can easily handle this one.

The developer tag is just one tag that is added to grabbable and touchable perms on creation of a new world:
![Image](https://github.com/user-attachments/assets/1301617c-6dd2-45e8-bfeb-22624e6ead51)

We can easily add additional default tags such as the one present on PF nodes. 

However, this is potentially a bug in the existing setup, because PF Visuals are meant to be protected anyway:
![Image](https://github.com/user-attachments/assets/a4c373f7-55ed-4b30-b0a3-6cba4171054d)


Eitherway, I consider this one to be no biggie, that's why I assigned it to myself yesterday, before I noticed Froox had it. I'm confident we'll find a fix.

**Comment by SilvyPaws** [comment]
> We can easily handle this one.
> 
> The developer tag is just one tag that is added to grabbable and touchable perms on creation of a new world: ![Image](https://github.com/user-attachments/assets/1301617c-6dd2-45e8-bfeb-22624e6ead51)
> 
> We can easily add additional default tags such as the one present on PF nodes.
> 
> However, this is potentially a bug in the existing setup, because PF Visuals are meant to be protected anyway: ![Image](https://github.com/user-attachments/assets/a4c373f7-55ed-4b30-b0a3-6cba4171054d)
> 
> Eitherway, I consider this one to be no biggie, that's why I assigned it to myself yesterday, before I noticed Froox had it. I dont want to step on his toes so I'll leave it to him but I'm confident we'll find a fix.

I'd like his opinion on this then on what he was planning to do (unless it was to address the bug you mentioned, because I didn't know any of that till you said it haha)

**Comment by shiftyscales** [comment]
To clarify based on @ProbablePrime's observation above- are you able to interact with the ProtoFlux node in any capacity when you aren't a builder, @SilvyPaws? Or is it specifically just grabbing that's the problem?

As far as I know, "Touchable" would just cover the interaction half of it, @ProbablePrime - and I didn't see explicit confirmation yet the nodes could be directly interacted with in any way other than being grabbed- hence I don't think there's a bug?

**Comment by ProbablePrime** [comment]
Good catch Shifty.

Grabbable Permissions doesn't have any types in it, so it has different settings to "touch" ![Image](https://github.com/user-attachments/assets/ae781b8b-4cc6-4ebe-9625-ab7901013fd6)

**Comment by SilvyPaws** [comment]
The contents can't be interacted with (i.e a string node's text content), the problem lies with being able to grab and delete them

**Comment by JackTheFoxOtter** [comment]
The Grabbable component is also on the node's slot itself, not on the visual, and that slot isn't tagged.
